mock/environment/__init__.py

```diff
@@ -27,3 +59,9 @@ def get_actual_conditions_variant(self,
             wind_sep=Angle(np.full(night_length, 40.0), unit='deg'),
             wind_spd=Quantity(np.full(night_length, 5.0 * u.m / u.s))
         )
+    
+    def data_file_path(self, filename: str) -> str:
```

Comment:
For this, you can make is a static method since it doesn't depend on anything in the class.

You can mark it:
```
@staticmethod
def data_file_path(filename: str) -> str:
    # ...
```
(It lets you get rid of the `self` since you don't need anything from `self`.)

You can then change the call up there from the way you have it:
```
input_filename = self.data_file_path(f'{site_lc}_weather_data.pickle.bz2')
```

to:

```
input_filename = Env.data_file_path(f'{site_lc}_weather_data.pickle.bz2')
```

Reply:
@stroncod `data_file_path` is a method that uses `os.path.join`. 😄